token/validator.go

```diff
@@ -43,8 +43,8 @@ func (c *Validator) UnmarshallAndVerify(ctx context.Context, ledger Ledger, anch
 
 // UnmarshallAndVerifyWithMetadata behaves as UnmarshallAndVerify. In addition, it returns the metadata extracts from the token request
 // in the form of map.
-func (c *Validator) UnmarshallAndVerifyWithMetadata(ctx context.Context, ledger Ledger, anchor string, raw []byte) ([]interface{}, map[string][]byte, error) {
-	actions, meta, err := c.backend.VerifyTokenRequestFromRaw(ctx, ledger.GetState, anchor, raw)
+func (c *Validator) UnmarshallAndVerifyWithMetadata(ctx context.Context, getState driver.GetStateFnc, anchor string, raw []byte) ([]interface{}, map[string][]byte, error) {
```

Comment:
I would prefer as it was because it is easier in the future to extend it with additional functions, if needed. Wdyt?